core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1178,6 +1178,37 @@ abstract class RDD[T: ClassTag](
     }
   }
 
+  /**
+   * Computing desired tree aggregate depth necessary to avoid exceeding
+   * driver.MaxResultSize during aggregation.
+   * Based on the formulae: (numPartitions)^(1/depth) * objectSize <= DriverMaxResultSize
+   * @param aggregatedObjectSizeInMb the size, in megabytes, of the object being tree aggregated
+   * @param numPartitions the number of partitions for which to aggregate partial results
+   * @param maxDriverResultSizeInMb the value of the parameter spark.driver.maxResultSize in MB
+   */
+  def getTreeAggregateIdealDepth(
+      aggregatedObjectSizeInMb: Int,
+      numPartitions: Int = getNumPartitions,
+      maxDriverResultSizeInMb: Int = Utils.memoryStringToMb(
+        context.getConf.get("spark.driver.maxResultSize"))): Int = {
+    require(
```

Comment:
I wouldn't check this condition here

Reply:
I though it could avoid users loosing time by tryiing to do an aggregation that is doomed to fail...
But I can remove it if you feel it's better to let the user get a "total size of results exceeded driver.maxResultSize" error.